consumer/consumererror/README.md

```diff
@@ -0,0 +1,264 @@
+# Consumer errors
+
+This package contains error types that should be returned by a consumer when an
+error occurs while processing telemetry. The error types included in this
+package provide functionality for communicating details about
+the error for use upstream in the pipeline. Ideally the error returned by a
+component in its `consume` function should be from this package.
+
+## Error classes
+
+**Retryable**: Errors are retryable if re-submitting data to a sink may result
+in a successful submission.
+
+**Permanent**: Errors are permanent if submission will always fail for the
+current data. Errors are considered permanent unless they are explicitly marked
+as retryable.
+
+## Use cases
+
+**Retry logic**: Errors should be allowed to include information necessary to
+perform retries.
+
+**Indicating partial success**: Errors can indicate that not all items were
+accepted, for example as in an OTLP partial success message. OTLP backends
+will return failed item counts if a partial success occurs, and this information
+can be propagated up to a receiver and returned to the caller.
+
+**Communicating network error codes**: Errors should allow embedding information
+necessary for the Collector to act as a proxy for a backend, i.e. relay a status
+code returned from a backend in a response to a system upstream from the
+Collector.
+
+## Current targets for using errors
+
+**Receivers**: Receivers should be able to consume multiple errors downstream
+and determine the best course of action based on the user's configuration. This
+may entail either keeping the retry queue inside of the Collector by having the
+receiver keep track of retries, or may involve having the caller manage the
+retry queue by returning a retryable network code with relevant information.
+
+**scraperhelper**: The scraper helper can use information about errors from
+downstream to affect scraping. For example, if the backend is struggling with
+the volume of data, scraping could be slowed, or the amount of data collected
+could be reduced. 
+
+**exporterhelper**: When an exporter returns a retryable error, the
+exporterhelper can use this information to manage the sending queue if it is
+enabled. Permanent errors will be forwarded back up the pipeline.
```

Comment:
My intention with this comment was that we would use these errors instead of the `throttleRetry` type, but I'm fine not calling out the sending queue specifically here.